src/services/SupabaseDataService.ts

```diff
@@ -29,6 +31,48 @@ export class SupabaseDataService extends BaseSupabaseService<KyselyDataDatabase>
     this.supabaseData = supabaseData;
   }
 
+  async mintBlueprintAndSwapInCollections(
```

Comment:
My expectation would be that this is an transaction so that these steps are handled atomically

Reply:
Resolved in https://github.com/hypercerts-org/hypercerts-api/pull/177/commits/194cb7330f09d8396b50450b7bfcb8734b104212